packages/discovery-provider/src/api/v1/models/notifications.py

```diff
@@ -1,37 +1,861 @@
 from flask_restx import fields
 
 from .common import ns
+from .extensions.fields import NestedOneOf, OneOfModel
 
 
 class Any(fields.Raw):
     def format(self, value):
         return value
 
 
-notification_action = ns.model(
+notification_base = ns.model(
+    "notification",
+    {
+        "type": fields.String(required=True),
+        "group_id": fields.String(required=True),
+        "is_seen": fields.Boolean(required=True),
+        "seen_at": fields.Integer(required=False),
+    },
+)
+
+
+notification_action_base = ns.model(
     "notification_action",
     {
         "specifier": fields.String(required=True),
+        "type": fields.String(required=True),
         "timestamp": fields.Integer(required=True),
-        "data": Any,
     },
 )
 
-notification = ns.model(
-    "notification",
+
```

Comment:
I cannot believe you did this. Also I cannot believe we have so many types and there isn't an easier way to generate all of them :-(
Thank you.